source/simulator/initial_conditions.cc

```diff
@@ -214,6 +91,55 @@ namespace aspect
 
         std::vector<types::global_dof_index> local_dof_indices (finite_element.dofs_per_cell);
 
+#if DEAL_II_VERSION_GTE(8,5,0)
```

Comment:
Yes, I did check that the changed code worked for 8.5.0-pre. I will check again with the most recent version of deal.II though.